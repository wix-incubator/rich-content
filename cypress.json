{
  "chromeWebSecurity": false,
  "baseUrl": "http://localhost:3002",
  "testFiles": "**/*.e2e.js",
  "ignoreTestFiles": [
    "**/__snapshots__/*",
    "**/__image_snapshots__/*"
  ],
  "integrationFolder": "e2e",
  "fixturesFolder": "e2e/cypress/fixtures",
  "pluginsFile": "e2e/cypress/plugins/index.js",
  "screenshotsFolder": "e2e/cypress/screenshots",
  "supportFile": "e2e/cypress/support/index.js",
  "video": true,
  "videoUploadOnPasses": false,
  "videosFolder": "e2e/cypress/videos",
  "env": {},
  "projectId": "6yj6sr"
}
