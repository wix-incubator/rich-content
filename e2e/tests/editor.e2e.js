import { INLINE_TOOLBAR_BUTTONS } from '../cypress/dataHooks';

describe('editor', () => {
  beforeEach(() => cy.switchToDesktop());

  it('should allow to enter text', () => {
    cy.loadEditor()
      .enterParagraphs([
        'Leverage agile frameworks',
        'to provide a robust synopsis for high level overviews.',
      ])
      .blurEditor()
      .matchSnapshots();
  });

  it('should allow to apply inline styles', () => {
    cy.loadEditor('plain')
      .setTextStyle(INLINE_TOOLBAR_BUTTONS.BOLD, [0, 5])
      .setTextStyle(INLINE_TOOLBAR_BUTTONS.UNDERLINE, [10, 5])
      .setTextStyle(INLINE_TOOLBAR_BUTTONS.ITALIC, [20, 5])
      .setTextStyle(INLINE_TOOLBAR_BUTTONS.BOLD, [30, 5])
      .setTextStyle(INLINE_TOOLBAR_BUTTONS.UNDERLINE)
      .setTextStyle(INLINE_TOOLBAR_BUTTONS.ITALIC)
      .setTextStyle(INLINE_TOOLBAR_BUTTONS.ALIGNMENT)
      .setTextStyle(INLINE_TOOLBAR_BUTTONS.TEXT_ALIGN_CENTER)
      .setTextStyle(INLINE_TOOLBAR_BUTTONS.QUOTE, [30, 170])
      .setColor(3, [20, 30])
      .setLineSpacing(3, [30, 100])
      .setSelection(0, 0)
      .blurEditor()
      .matchSnapshots();
  });

  it('should allow to create lists', () => {
    cy.loadEditor('plain')
      .setTextStyle(INLINE_TOOLBAR_BUTTONS.ORDERED_LIST, [300, 100])
      .setTextStyle(INLINE_TOOLBAR_BUTTONS.UNORDERED_LIST, [550, 1])
      .matchSnapshots();
  });
});
