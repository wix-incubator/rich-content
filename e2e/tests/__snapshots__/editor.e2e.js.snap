exports[`editor > should allow to enter text #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "Leverage agile frameworks",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 1,
      "text": "to provide a robust synopsis for high level overviews.",
      "type": "unstyled"
    }
  ],
  "entityMap": {}
};

exports[`editor > should allow to apply inline styles #0`] =
{
  "blocks": [
    {
      "data": {
        "dynamicStyles": {
          "line-height": "2.5",
          "padding-bottom": "3px",
          "padding-top": "2px"
        },
        "textAlignment": "center"
      },
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [
        {
          "length": 5,
          "offset": 0,
          "style": "BOLD"
        },
        {
          "length": 5,
          "offset": 30,
          "style": "BOLD"
        },
        {
          "length": 5,
          "offset": 10,
          "style": "UNDERLINE"
        },
        {
          "length": 5,
          "offset": 30,
          "style": "UNDERLINE"
        },
        {
          "length": 5,
          "offset": 20,
          "style": "ITALIC"
        },
        {
          "length": 5,
          "offset": 30,
          "style": "ITALIC"
        },
        {
          "length": 30,
          "offset": 20,
          "style": "color4"
        }
      ],
      "key": 0,
      "text": "Leverage agile frameworks to provide a robust synopsis for high level overviews. Iterative approaches to corporate strategy foster collaborative thinking to further the overall value proposition. ",
      "type": "blockquote"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 1,
      "text": "Organically grow the holistic world view of disruptive innovation via workplace diversity and empowerment.",
      "type": "blockquote"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 2,
      "text": "Bring to the table win-win survival strategies to ensure proactive domination.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 3,
      "text": "At the end of the day, going forward, a new normal that has evolved from generation X is on the runway heading towards a streamlined cloud solution. ",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 4,
      "text": "User generated content in real-time will have multiple touchpoints for offshoring.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 5,
      "text": "Capitalize on low hanging fruit to identify a ballpark value added activity to beta test. ",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 6,
      "text": "Override the digital divide with additional clickthroughs from DevOps.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 7,
      "text": "Nanotechnology immersion along the information highway will close the loop on focusing solely on the bottom line.",
      "type": "unstyled"
    }
  ],
  "entityMap": {}
};

exports[`editor > should allow to create lists #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "Leverage agile frameworks to provide a robust synopsis for high level overviews. Iterative approaches to corporate strategy foster collaborative thinking to further the overall value proposition. ",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 1,
      "text": "Organically grow the holistic world view of disruptive innovation via workplace diversity and empowerment.",
      "type": "ordered-list-item"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 2,
      "text": "Bring to the table win-win survival strategies to ensure proactive domination.",
      "type": "ordered-list-item"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 3,
      "text": "At the end of the day, going forward, a new normal that has evolved from generation X is on the runway heading towards a streamlined cloud solution. ",
      "type": "ordered-list-item"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 4,
      "text": "User generated content in real-time will have multiple touchpoints for offshoring.",
      "type": "unordered-list-item"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 5,
      "text": "Capitalize on low hanging fruit to identify a ballpark value added activity to beta test. ",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 6,
      "text": "Override the digital divide with additional clickthroughs from DevOps.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 7,
      "text": "Nanotechnology immersion along the information highway will close the loop on focusing solely on the bottom line.",
      "type": "unstyled"
    }
  ],
  "entityMap": {}
};

exports[`editor > when in hebrew locale > toolbar should render rtl #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "Leverage agile frameworks to provide a robust synopsis for high level overviews. Iterative approaches to corporate strategy foster collaborative thinking to further the overall value proposition. ",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 1,
      "text": "Organically grow the holistic world view of disruptive innovation via workplace diversity and empowerment.",
      "type": "ordered-list-item"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 2,
      "text": "Bring to the table win-win survival strategies to ensure proactive domination.",
      "type": "ordered-list-item"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 3,
      "text": "At the end of the day, going forward, a new normal that has evolved from generation X is on the runway heading towards a streamlined cloud solution. ",
      "type": "ordered-list-item"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 4,
      "text": "User generated content in real-time will have multiple touchpoints for offshoring.",
      "type": "unordered-list-item"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 5,
      "text": "Capitalize on low hanging fruit to identify a ballpark value added activity to beta test. ",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 6,
      "text": "Override the digital divide with additional clickthroughs from DevOps.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 7,
      "text": "Nanotechnology immersion along the information highway will close the loop on focusing solely on the bottom line.",
      "type": "unstyled"
    }
  ],
  "entityMap": {}
};

exports[`editor > when in hebrew locale > should render text toolbar in rtl #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "Leverage agile frameworks to provide a robust synopsis for high level overviews. Iterative approaches to corporate strategy foster collaborative thinking to further the overall value proposition. ",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 1,
      "text": "Organically grow the holistic world view of disruptive innovation via workplace diversity and empowerment.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 2,
      "text": "Bring to the table win-win survival strategies to ensure proactive domination.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 3,
      "text": "At the end of the day, going forward, a new normal that has evolved from generation X is on the runway heading towards a streamlined cloud solution. ",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 4,
      "text": "User generated content in real-time will have multiple touchpoints for offshoring.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 5,
      "text": "Capitalize on low hanging fruit to identify a ballpark value added activity to beta test. ",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 6,
      "text": "Override the digital divide with additional clickthroughs from DevOps.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 7,
      "text": "Nanotechnology immersion along the information highway will close the loop on focusing solely on the bottom line.",
      "type": "unstyled"
    }
  ],
  "entityMap": {}
};

exports[`editor > when in hebrew locale > should render rtl and ltr text correctly #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "לורם איפסום דולור סיט אמט, קונסקטורר אדיפיסינג אלית נולום ארווס סאפיאן - פוסיליס קוויס, אקווזמן צש בליא, מנסוטו צמלח לביקו ננבי, צמוקו בלוקריה שיצמה ברורק. ושבעגט ליבם סולגק. בראיט ולחת צורק מונחף, בגורמי מגמש. תרבנך וסתעד לכנו סתשם השמה - לתכי מורגם בורק? לתיג ישבעס.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 1,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 2,
      "text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
      "type": "unstyled"
    }
  ],
  "entityMap": {}
};

exports[`editor > when in hebrew locale > should render plugin toolbar in rtl #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "",
      "type": "unstyled"
    }
  ],
  "entityMap": {}
};

exports[`editor > when in hebrew locale > should render external modal in rtl #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "Leverage agile frameworks to provide a robust synopsis for high level overviews. Iterative approaches to corporate strategy foster collaborative thinking to further the overall value proposition. ",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 1,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [
        {
          "key": 0,
          "length": 1,
          "offset": 0
        }
      ],
      "inlineStyleRanges": [],
      "key": 2,
      "text": " ",
      "type": "atomic"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 3,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 4,
      "text": "Organically grow the holistic world view of disruptive innovation via workplace diversity and empowerment.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 5,
      "text": "Bring to the table win-win survival strategies to ensure proactive domination.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 6,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [
        {
          "key": 1,
          "length": 1,
          "offset": 0
        }
      ],
      "inlineStyleRanges": [],
      "key": 7,
      "text": " ",
      "type": "atomic"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 8,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 9,
      "text": "At the end of the day, going forward, a new normal that has evolved from generation X is on the runway heading towards a streamlined cloud solution.",
      "type": "unstyled"
    }
  ],
  "entityMap": {
    "0": {
      "data": {
        "config": {
          "alignment": "center",
          "showDescription": true,
          "showTitle": true,
          "size": "small"
        },
        "src": {
          "file_name": "8bb438_131a7e1872bc45ec827bb61e56b840fe.jpg",
          "height": 3354,
          "id": "036c6bf6cef5e4409848eb4eb6f80de1",
          "original_file_name": "8bb438_131a7e1872bc45ec827bb61e56b840fe.jpg",
          "width": 2898
        }
      },
      "mutability": "IMMUTABLE",
      "type": "wix-draft-plugin-image"
    },
    "1": {
      "data": {
        "config": {
          "alignment": "center",
          "showDescription": true,
          "showTitle": true,
          "size": "content"
        },
        "src": {
          "file_name": "8bb438_4af301c080294224b6b5e15cd38a035f.jpg",
          "height": 1280,
          "id": "759c83479429d725f03367897235a8b5",
          "original_file_name": "8bb438_4af301c080294224b6b5e15cd38a035f.jpg",
          "width": 1920
        }
      },
      "mutability": "IMMUTABLE",
      "type": "wix-draft-plugin-image"
    }
  }
};

exports[`editor > when in hebrew locale > desktop > should render plugin toolbar in rtl #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "",
      "type": "unstyled"
    }
  ],
  "entityMap": {}
};

exports[`editor > when in hebrew locale > desktop > should render text toolbar in rtl #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "Leverage agile frameworks to provide a robust synopsis for high level overviews. Iterative approaches to corporate strategy foster collaborative thinking to further the overall value proposition. ",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 1,
      "text": "Organically grow the holistic world view of disruptive innovation via workplace diversity and empowerment.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 2,
      "text": "Bring to the table win-win survival strategies to ensure proactive domination.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 3,
      "text": "At the end of the day, going forward, a new normal that has evolved from generation X is on the runway heading towards a streamlined cloud solution. ",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 4,
      "text": "User generated content in real-time will have multiple touchpoints for offshoring.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 5,
      "text": "Capitalize on low hanging fruit to identify a ballpark value added activity to beta test. ",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 6,
      "text": "Override the digital divide with additional clickthroughs from DevOps.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 7,
      "text": "Nanotechnology immersion along the information highway will close the loop on focusing solely on the bottom line.",
      "type": "unstyled"
    }
  ],
  "entityMap": {}
};

exports[`editor > when in hebrew locale > desktop > should render rtl and ltr text correctly #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "לורם איפסום דולור סיט אמט, קונסקטורר אדיפיסינג אלית נולום ארווס סאפיאן - פוסיליס קוויס, אקווזמן צש בליא, מנסוטו צמלח לביקו ננבי, צמוקו בלוקריה שיצמה ברורק. ושבעגט ליבם סולגק. בראיט ולחת צורק מונחף, בגורמי מגמש. תרבנך וסתעד לכנו סתשם השמה - לתכי מורגם בורק? לתיג ישבעס.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 1,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 2,
      "text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
      "type": "unstyled"
    }
  ],
  "entityMap": {}
};

exports[`editor > when in hebrew locale > desktop > should render external modal in rtl #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "Leverage agile frameworks to provide a robust synopsis for high level overviews. Iterative approaches to corporate strategy foster collaborative thinking to further the overall value proposition. ",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 1,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [
        {
          "key": 0,
          "length": 1,
          "offset": 0
        }
      ],
      "inlineStyleRanges": [],
      "key": 2,
      "text": " ",
      "type": "atomic"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 3,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 4,
      "text": "Organically grow the holistic world view of disruptive innovation via workplace diversity and empowerment.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 5,
      "text": "Bring to the table win-win survival strategies to ensure proactive domination.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 6,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [
        {
          "key": 1,
          "length": 1,
          "offset": 0
        }
      ],
      "inlineStyleRanges": [],
      "key": 7,
      "text": " ",
      "type": "atomic"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 8,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 9,
      "text": "At the end of the day, going forward, a new normal that has evolved from generation X is on the runway heading towards a streamlined cloud solution.",
      "type": "unstyled"
    }
  ],
  "entityMap": {
    "0": {
      "data": {
        "config": {
          "alignment": "center",
          "showDescription": true,
          "showTitle": true,
          "size": "small"
        },
        "src": {
          "file_name": "8bb438_131a7e1872bc45ec827bb61e56b840fe.jpg",
          "height": 3354,
          "id": "036c6bf6cef5e4409848eb4eb6f80de1",
          "original_file_name": "8bb438_131a7e1872bc45ec827bb61e56b840fe.jpg",
          "width": 2898
        }
      },
      "mutability": "IMMUTABLE",
      "type": "wix-draft-plugin-image"
    },
    "1": {
      "data": {
        "config": {
          "alignment": "center",
          "showDescription": true,
          "showTitle": true,
          "size": "content"
        },
        "src": {
          "file_name": "8bb438_83e03311e53a47ce9ba8bda910fdad88.jpg",
          "height": 3526,
          "id": "b12907da5fbb55a31a760e49fb0bb4e6",
          "original_file_name": "8bb438_83e03311e53a47ce9ba8bda910fdad88.jpg",
          "width": 5177
        }
      },
      "mutability": "IMMUTABLE",
      "type": "wix-draft-plugin-image"
    }
  }
};

exports[`editor > when in hebrew locale > mobile > should render plugin toolbar in rtl #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "",
      "type": "unstyled"
    }
  ],
  "entityMap": {}
};

exports[`editor > when in hebrew locale > mobile > should render rtl and ltr text correctly #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "לורם איפסום דולור סיט אמט, קונסקטורר אדיפיסינג אלית נולום ארווס סאפיאן - פוסיליס קוויס, אקווזמן צש בליא, מנסוטו צמלח לביקו ננבי, צמוקו בלוקריה שיצמה ברורק. ושבעגט ליבם סולגק. בראיט ולחת צורק מונחף, בגורמי מגמש. תרבנך וסתעד לכנו סתשם השמה - לתכי מורגם בורק? לתיג ישבעס.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 1,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 2,
      "text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
      "type": "unstyled"
    }
  ],
  "entityMap": {}
};

exports[`editor > when in hebrew locale > mobile > should render external modal in rtl #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "Leverage agile frameworks to provide a robust synopsis for high level overviews. Iterative approaches to corporate strategy foster collaborative thinking to further the overall value proposition. ",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 1,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [
        {
          "key": 0,
          "length": 1,
          "offset": 0
        }
      ],
      "inlineStyleRanges": [],
      "key": 2,
      "text": " ",
      "type": "atomic"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 3,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 4,
      "text": "Organically grow the holistic world view of disruptive innovation via workplace diversity and empowerment.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 5,
      "text": "Bring to the table win-win survival strategies to ensure proactive domination.",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 6,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [
        {
          "key": 1,
          "length": 1,
          "offset": 0
        }
      ],
      "inlineStyleRanges": [],
      "key": 7,
      "text": " ",
      "type": "atomic"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 8,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 9,
      "text": "At the end of the day, going forward, a new normal that has evolved from generation X is on the runway heading towards a streamlined cloud solution.",
      "type": "unstyled"
    }
  ],
  "entityMap": {
    "0": {
      "data": {
        "config": {
          "alignment": "center",
          "showDescription": true,
          "showTitle": true,
          "size": "small"
        },
        "src": {
          "file_name": "8bb438_131a7e1872bc45ec827bb61e56b840fe.jpg",
          "height": 3354,
          "id": "036c6bf6cef5e4409848eb4eb6f80de1",
          "original_file_name": "8bb438_131a7e1872bc45ec827bb61e56b840fe.jpg",
          "width": 2898
        }
      },
      "mutability": "IMMUTABLE",
      "type": "wix-draft-plugin-image"
    },
    "1": {
      "data": {
        "config": {
          "alignment": "center",
          "showDescription": true,
          "showTitle": true,
          "size": "content"
        },
        "src": {
          "file_name": "8bb438_83e03311e53a47ce9ba8bda910fdad88.jpg",
          "height": 3526,
          "id": "b12907da5fbb55a31a760e49fb0bb4e6",
          "original_file_name": "8bb438_83e03311e53a47ce9ba8bda910fdad88.jpg",
          "width": 5177
        }
      },
      "mutability": "IMMUTABLE",
      "type": "wix-draft-plugin-image"
    }
  }
};

exports[`editor > when in hebrew locale > mobile > should render add plugin modal in rtl #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "",
      "type": "unstyled"
    }
  ],
  "entityMap": {}
};
